### PR TITLE
[ci][port/01] make pipeline id in rayci configurable

### PR DESCRIPTION
**Pull request by can-anyscale**
Make pipeline configuration in rayci so we can support it across repos

Test:
- CI
- e2e uploading: https://buildkite.com/ray-project/postmerge/builds/3713

### CODE REVIEW

File: release/ray_release/configs/global_config.py [hunk 1]
```diff
@@ -48,6 +51,8 @@ def _init_global_config(config_file: str):
         byod_gcp_cr=config_content["byod"].get("gcp_cr"),
         state_machine_aws_bucket=config_content["state_machine"]["aws_bucket"],
         aws2gce_credentials=config_content.get("credentials", {}).get("aws2gce"),
+        pipeline_premerge=config_content.get("pipeline", {}).get("premerge", []),
+        pipeline_postmerge=config_content.get("pipeline", {}).get("postmerge", []),
```
Comment (aslonnie) [inline release/ray_release/configs/global_config.py:55]: maybe call this `premerge_pipeline_ids`?

and maybe check that a pipeline is uniquely either `premerge` or `postmerge`?

File: release/ray_release/configs/global_config.py [hunk 2]
```diff
@@ -13,6 +14,8 @@ class GlobalConfig(TypedDict):
     byod_gcp_cr: str
     state_machine_aws_bucket: str
     aws2gce_credentials: str
+    pipeline_premerge: List[str]
+    pipeline_postmerge: List[str]
```
Comment (aslonnie) [inline release/ray_release/configs/global_config.py:18]: nit: maybe make these sets?

**Review by aslonnie — COMMENTED** [review]
- I am still not sure how this will work downstream
- and I am also a bit uncomfortable that premerge/postmerge concept is in a config file under `ray_release` directory.

how urgent is this? can those issues be fixed?

**Comment by can-anyscale** [comment]
yeah no rush, let merge the release directory into ci directory first ;)

**Comment by can-anyscale** [comment]
Let me write a doc first on a plan to unify the CI/CD across multiple repositories; perhaps that will make the implementation easier to understand going forward

**Pull request closed without merging**